### ISSUE TITLE
plugin/template: cannot use handler as type plugin.Handler in return argument

### ISSUE THREAD

**Issue by greenpau** [coredns/coredns#1466]
Test fails:

```
$ go test
# github.com/coredns/coredns/plugin/template
plugin/template/setup.go:33: cannot use handler (type Handler) as type plugin.Handler in return argument:
	Handler does not implement plugin.Handler (wrong type for ServeDNS method)
		have ServeDNS("context".Context, dns.ResponseWriter, *dns.Msg) (int, error)
		want ServeDNS("golang.org/x/net/context".Context, dns.ResponseWriter, *dns.Msg) (int, error)
```

**Comment by fturib** [comment]
I think I had the same .. and was resolved when I moved to go 1.9.2

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "[coredns/coredns] plugin/template: ..." ]
>Test fails:
>
>```
>$ go test
># github.com/coredns/coredns/plugin/template
>plugin/template/setup.go:33: cannot use handler (type Handler) as type plugin.Handler in return argument:
>	Handler does not implement plugin.Handler (wrong type for ServeDNS method)
>		have ServeDNS("context".Context, dns.ResponseWriter, *dns.Msg) (int, error)
>		want ServeDNS("golang.org/x/net/context".Context, dns.ResponseWriter, *dns.Msg) (int, error)
>```

template imports the wrong "context" package :-(
